src/main/java/com/cisco/trex/stateful/api/lowlevel/ASTFClientTemplate.java

```diff
@@ -7,32 +7,26 @@
  * Template class
  */
 abstract class ASTFClientTemplate extends ASTFTemplateBase {
-  private ASTFCluster astfCluster;
-  private ASTFIpGen iPGen;
+  private ASTFCluster cluster;
+  private ASTFIpGen ipGen;
 
-  /**
-   * construct
-   *
-   * @param iPGen
-   * @param astfCluster
-   * @param astfProgram
-   */
-  public ASTFClientTemplate(ASTFIpGen iPGen, ASTFCluster astfCluster, ASTFProgram astfProgram) {
-    super(astfProgram);
-    this.iPGen = iPGen;
-    this.astfCluster = astfCluster == null ? new ASTFCluster() : astfCluster;
+  public ASTFClientTemplate(ASTFIpGen ipGen, ASTFCluster cluster, ASTFProgram program) {
+    super(program);
+    fields.add("ip_gen", ipGen.toJson());
```

Comment:
from the python code:
```
    def __init__(self, ip_gen, cluster=ASTFCluster(), program=None):
        super(_ASTFClientTemplate, self).__init__(program=program)
        self.fields['ip_gen'] = ip_gen
        self.fields['cluster'] = cluster

    def to_json(self):
        ret = super(_ASTFClientTemplate, self).to_json()
        ret['ip_gen'] = self.fields['ip_gen'].to_json()
        ret['cluster'] = self.fields['cluster'].to_json()
        return ret
```
the fields["ip_gen"]& field['cluster'] are only used to generate json, but in this java toJson method, it uses ipGen.toJson() & cluster.toJson() to do the similar as python code.
so it seems no need to add these two into fields map. please double check it.

Reply:
I think they are needed, for example the filed ip_gen , it's used in the class ASTFProfileCache .